pkg/messaging/nats/pubsub.go

```diff
@@ -143,27 +169,35 @@ func (ps *pubsub) Unsubscribe(ctx context.Context, id, topic string) error {
 			return err
 		}
 	}
-	if err := current.Unsubscribe(); err != nil {
+
+	current.Stop()
+
+	if err := ps.stream.DeleteConsumer(ctx, id); err != nil {
```

Comment:
Unlike plain NATS, JS client is able to hold consumer (subscription) data. Do we need to hold our own list of subscriptions in that case?